### DISCUSSION TITLE
Video - Almost Ready for Install into Gloves

### DISCUSSION THREAD

**Discussion by Neurosmith24** [kriswilk/buzzah#13] [General]
Question for anyone who has built these so far. I have a basic functioning prototype with the electrical components and LRAs installed in housing. I did a test. I was a little surprised to see how much the housing vibrate - I was hoping my foam padding would dampen it to just the contact point. I shot a video of it to show what I mean. Does this seem normal? @PWPInnovator898 does this match what you've built? 

Another way to ask it, I have this going and touch the bottom/underneath portion of the housing, not where the finger goes, I can still feel a vibration. Does that make it too general of a vibration to work? Should it be isolated more? When wearing the real Stanford gloves can vibrations be felt anywhere else on the tactor housing aside from the contact point with the fingertip?

https://github.com/kriswilk/buzzah/assets/5769522/afdaef2a-d8ac-4b12-8626-230a26f5ef24

**Comment by Neurosmith24** [comment]
Here's a closer photo of the housings if anyone is curious about them. Not super thrilled with the quality of the 3D prints from the local library. 
<img width="694" alt="Screenshot 2024-03-15 at 2 28 14 PM" src="https://github.com/kriswilk/buzzah/assets/5769522/d4b984ce-bdee-4594-b70d-4025125bfabb">

**Comment by kriswilk** [comment]
I haven't built a complete set but imagine that this is to be expected. Physics is physics... If the business end of the LRA vibrates in one direction, the body will necessarily move in the other direction. Once firmly attached to a finger (a much larger mass) I suspect the vibration of the case will be reduced.

That's all assuming the LRA has a freely-moving external element. If instead it has a sealed case (i.e. the vibrating element is internal) then it might be more difficult to localize the vibration.

**Reply by PWPInnovator898** [reply]
I agree re: the physics of the LRAs. We've struggled to isolate the vibration, too.  Foam inserts in the housings have been the best way to do this for the LRAs.  Yet, this still is not optimal, as we may be over stimulating,  For that reason, I'm considering doing more testing of audio exciters, as others have done, to better isolate the vibration.

**Reply by PWPInnovator898** [reply]
BTW, audio exciters can be driven w/ the current Buzzah electronics. The TI haptic IC uses a PWM-modulated, Class D output driver.
re: your Stanford gloves question: I will have to request testing them again from the early trial participant, since I did not notice if vibrations could be be felt anywhere else on the tactor housing aside from the contact point with the fingertip.  However, I do remember the very distinct sensation of the contactor movement into and away from the fingertips, and less of a general vibration from LRA-based housing designs.   The biggest challenge remains calibrating the stimulation settings without some kind of neurofeedback.  

Dr. Tass referred to this recently in his latest webinar. 
https://youtu.be/_gmdfrkW2oA?si=6djVdT__ay5eiKNR
To save yourself some time, the bulk of the webinar is likely familiar to you. 
Key takeaways:
1- go to ~1 hr 9 min for Q&A session.
He commented that the original tactors were too expensive and they’ve now been replaced by much simpler TACTORS  that are just a couple of bucks each.  They  have switched to the cheaper versions in the latest gloves which they’re starting to do their pilot studies with.   
2-with the new gloves, they are now calibrating the stimulation on a per patient basis by measuring "brain activity".  He did not go into details.

Hope this helps.

**Reply by Neurosmith24** [reply]
Thanks, yes that helps. Good to know your experience may be closer to mine that I feared. Think I'll try some additional foam padding (using your high tech foam paint brush idea ).

I'm really interested now in the audio exciters. If Buzzah electronics as I have them can drive an audio exciter, then really it should just be an issue of buying the audio exciters and redesignin the housing to hold them. (BTW, no idea what a PWM-modulated Class D output means).

@kriswilk any additional input on the Buzzah elecronics working with audio exciters? Can I just wire up audio exciters to the Adafruit DRV2605L Haptic Motor Controller and edit the code to specify these drivers instead?

Thanks for your help guys!

**Reply by PWPInnovator898** [reply]
The TI driver doesn't have an "audio exciter" operating mode, but it's Class-D output can drive an audio exciter which "looks" like an audio speaker, with reasonably similar load characteristics (impedance) to LRAs. 
Notes:
- Some playing with the TI operating settings may be required to get optimal vibration  
- I have only tested with one exciter so far, pending more tests.
@kriswilk - your thoughts?

**Reply by Neurosmith24** [reply]
Thanks. FWIW, these are the latest audio exciters that WTP is using - specs included:
https://www.daytonaudio.com/product/1295/daex-9-4sm-haptic-feedback-and-audio-exciter-9mm-1w-4-ohm

**Reply by PWPInnovator898** [reply]
Yes, that’s the audio exciter I have in hand that I have tested preliminarily. Stimulation intensity is markedly lower than the LRA at the same drive level

**Reply by kriswilk** [reply]
The TI DRV2605L is designed to drive a load with an impedance of 8 Ohms. While it's true that the driver could operate other types of tactors/exciters, the one from Dayton Audio has an impedance of 4 Ohms. This is at the limit of the TI driver... in fact, the specs indicate that a load impedance at or below 4 Ohms will trigger the overcurrent shutdown safety feature.

That said, if it seems to work, great! But YMMV.

**Reply by Neurosmith24** [reply]
Thanks Kris. So, forgive me ignorance, but if we found an audio exciter with specs of 8 Ohms would that work? I assume there would be a min and max watts too? This one has an impedance of 8 Ohms and 2W: https://www.digikey.com/en/products/detail/pui-audio,-inc./ASX03308-SM-R/6125072?utm_adgroup=&utm_source=google&utm_medium=cpc&utm_campaign=PMax%20Shopping_Product_Low%20ROAS%20Categories&utm_term=&utm_content=&utm_id=go_cmp-20243063506_adg-_ad-__dev-c_ext-_prd-6125072_sig-Cj0KCQjwqdqvBhCPARIsANrmZhMUMudUtq51rOWWYOJWlQQDMQ39IsdTSdiCToLpvP0Wmt3ekGFO1hIaAj8GEALw_wcB&gad_source=1&gclid=Cj0KCQjwqdqvBhCPARIsANrmZhMUMudUtq51rOWWYOJWlQQDMQ39IsdTSdiCToLpvP0Wmt3ekGFO1hIaAj8GEALw_wcB

**Reply by PWPInnovator898** [reply]
That's a good point https://github.com/kriswilk , re: impedance of 4 Ohms and triggering the overcurrent shutdown.
https://github.com/Neurosmith24 - I'll look for other exciters that may work, but I'm concerned the vibration intensity I've observed from the 9mm Dayton exciter is very weak compared w/ the LRAs driven from the DRV2605L, and from what I remember feeling from the Stanford gloves.  Admittedly, that's just one datapoint for one pair gloves calibrated for one early trial participant. 
Also, re: the exciter noted from DigiKey, its  frequency response is not an ideal match, as it's spec'd for 420 Hz to ~10 kHz, whereas the nominal VCR stimulation frequency is 250 Hz. It should vibrate at 250 Hz, but at an unspecified intensity level.

**Reply by Neurosmith24** [reply]
What kind of Watt parameters would we need to keep within? I found a couple of options at 8 Ohms and 220 Hz. They're a bit larger than ideal but might be worth considering. They come in at 10W and 20W.

**Reply by PWPInnovator898** [reply]
I’ll continue to look for candidates for the audio exciter.   The power isn’t the issue it’s more the vibrational response. But moving to an audio exciter would change the driving electronics significantly.

**Reply by kriswilk** [reply]
Questions about actuator compatibility are probably best answered directly by TI. It might be worth asking in their technical discussion forum. But 10-20 W isn't possible. Assuming a drive voltage between 2-3 V and an impedance of 8 Ohms, the maximum power output would be roughly 1 W.

Many audio exciters and speakers in general may not be ideal because they have wide surface areas. Even at relatively high output power, the diaphragm may only move very, very slightly. An actuator with a small area and the capability to be driven back and forth with higher physical amplitude would be ideal.
